commit 4592411784ccf83f873f98ba94aeae2482783c9a
Author: Justin Stitt <justinstitt@google.com>
Date:   Mon Oct 23 20:26:13 2023 +0000

    scsi: csiostor: Replace deprecated strncpy() with strscpy()
    
    strncpy() is deprecated for use on NUL-terminated destination strings [1]
    and as such we should prefer more robust and less ambiguous string
    interfaces.
    
    'hw' is kzalloc'd just before this string assignment:
    |       hw = kzalloc(sizeof(struct csio_hw), GFP_KERNEL);
    
    ... which means any NUL-padding is redundant.
    
    Since CSIO_DRV_VERSION is a small string literal (smaller than
    sizeof(dest)):
    
    ... there is functionally no change in this swap from strncpy() to
    strscpy(). Nonetheless, let's make the change for robustness' sake -- as
    it will ensure that drv_version is _always_ NUL-terminated.
    
    Link: https://www.kernel.org/doc/html/latest/process/deprecated.html#strncpy-on-nul-terminated-strings [1]
    Link: https://github.com/KSPP/linux/issues/90
    Cc: linux-hardening@vger.kernel.org
    Signed-off-by: Justin Stitt <justinstitt@google.com>
    Link: https://lore.kernel.org/r/20231023-strncpy-drivers-scsi-csiostor-csio_init-c-v1-1-5ea445b56864@google.com
    Reviewed-by: Kees Cook <keescook@chromium.org>
    Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>

diff --git a/drivers/scsi/csiostor/csio_init.c b/drivers/scsi/csiostor/csio_init.c
index 0c32faefad7c..d649b7a2a879 100644
--- a/drivers/scsi/csiostor/csio_init.c
+++ b/drivers/scsi/csiostor/csio_init.c
@@ -521,7 +521,8 @@ static struct csio_hw *csio_hw_alloc(struct pci_dev *pdev)
 		goto err;
 
 	hw->pdev = pdev;
-	strncpy(hw->drv_version, CSIO_DRV_VERSION, 32);
+	strscpy(hw->drv_version, CSIO_DRV_VERSION,
+		sizeof(hw->drv_version));
 
 	/* memory pool/DMA pool allocation */
 	if (csio_resource_alloc(hw))